tools/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+# Create a header-only library from an SVD register definition file.
+#
+# add_groov_reg_lib(
+#     groov_reg_test 
+#     INPUT_FILE "rp2040.svd"
+#     OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}
+#     PERIPHERALS adc rtc
+# )
+
+find_package(Python3 COMPONENTS Interpreter)
+
+if(Python3_FOUND)
+    message(STATUS "Python3 Version: ${Python3_VERSION}")
+    message(STATUS "Python3 Executable: ${Python3_EXECUTABLE}")
+
+    function(add_groov_reg_lib target_name)
+        set(options "")
+        set(oneValueArgs INPUT_FILE OUTPUT_DIR)
+        set(multiValueArgs PERIPHERALS)
+        cmake_parse_arguments(PARSE_ARGV 1 ARG "${options}" "${oneValueArgs}" "${multiValueArgs}")
```

Comment:
**question**: Are you purposely alllowing the args to have semicolons in them? Or is there a reason not to use `cmake_parse_arguments(ARG "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})` ?

Reply:
I want a single positional argument, the first argument, to be the name of the library it will create.